app/invocation/invocation_action_card.tsx

```diff
@@ -553,7 +596,57 @@ export default class InvocationActionCardComponent extends React.Component<Props
               )}
               <div>
                 <div className="action-line">
-                  <div className="action-title">Result details</div>
+                  <div className="action-header">
+                    <div className="action-title">Result details</div>
+                    {vmMetadata?.snapshotKey && (
+                      <div className="invocation-menu-container">
+                        <OutlinedButton
+                          onClick={this.onClickShowMenu.bind(this)}
+                          className="invocation-menu-button"
+                          title="More result detail options">
+                          <MoreVertical />
+                        </OutlinedButton>
+                        <Popup isOpen={this.state.isMenuOpen} onRequestClose={this.onCloseMenu.bind(this)}>
+                          <Menu>
+                            <MenuItem onClick={() => this.setState({ showInvalidateSnapshotModal: true })}>
+                              Invalidate VM snapshot
+                            </MenuItem>
+                          </Menu>
+                        </Popup>
+                        <Modal
+                          isOpen={this.state.showInvalidateSnapshotModal}
+                          onRequestClose={() =>
+                            this.setState({ showInvalidateSnapshotModal: false, isMenuOpen: false })
+                          }>
+                          <Dialog>
+                            <DialogHeader>
+                              <DialogTitle>Confirm invalidate VM snapshot</DialogTitle>
+                            </DialogHeader>
+                            <DialogBody>
+                              <p>Are you sure you want to invalidate the recycled VM snapshot used for this action?</p>
+                              <p>
+                                That will prevent this recycled runner from being reused. This will make future actions
+                                slower, so this flag is not encouraged.
+                              </p>
```

Comment:
I don't think we need this warning just yet since these are normal actions rather than VM actions, but if you think it's worth keeping I think it could be clarified a little

```suggestion
                              <p>
                                A new VM will be booted for the next run of this action, which may result in longer execution time.
                              </p>
```